flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/GenericInMemoryCatalog.java

```diff
@@ -0,0 +1,319 @@
+			throw new DatabaseNotExistException(catalogName, databaseName);
+		} else {
+			return databases.get(databaseName).copy();
+		}
+	}
+
+	@Override
+	public boolean databaseExists(String databaseName) {
+		checkArgument(!StringUtils.isNullOrWhitespaceOnly(databaseName));
+
+		return databases.containsKey(databaseName);
+	}
+
+	// ------ tables ------
+
+	@Override
+	public void createTable(ObjectPath tablePath, CatalogTable table, boolean ignoreIfExists)
+		throws TableAlreadyExistException, DatabaseNotExistException {
+		checkArgument(tablePath != null);
+		checkArgument(table != null);
+
+		if (!databaseExists(tablePath.getDatabaseName())) {
+			throw new DatabaseNotExistException(catalogName, tablePath.getDatabaseName());
+		}
+
+		if (tableExists(tablePath)) {
+			if (!ignoreIfExists) {
+				throw new TableAlreadyExistException(catalogName, tablePath);
+			}
+		} else {
+			tables.put(tablePath, table.copy());
+		}
+	}
+
+	@Override
+	public void alterTable(ObjectPath tablePath, CatalogTable newTable, boolean ignoreIfNotExists)
+		throws TableNotExistException {
+		checkArgument(tablePath != null);
+		checkArgument(newTable != null);
+
+		if (tableExists(tablePath)) {
+			tables.put(tablePath, newTable.copy());
+		} else if (!ignoreIfNotExists) {
+			throw new TableNotExistException(catalogName, tablePath);
+		}
+	}
+
+	// ------ tables and views ------
+
+	@Override
+	public void dropTable(ObjectPath tablePath, boolean ignoreIfNotExists) throws TableNotExistException {
+		checkArgument(tablePath != null);
+
+		if (tableExists(tablePath)) {
+			tables.remove(tablePath);
+		} else if (!ignoreIfNotExists) {
+			throw new TableNotExistException(catalogName, tablePath);
+		}
+	}
+
+	@Override
+	public void renameTable(ObjectPath tablePath, String newTableName, boolean ignoreIfNotExists)
+		throws TableNotExistException, TableAlreadyExistException {
+		checkArgument(tablePath != null);
+		checkArgument(!StringUtils.isNullOrWhitespaceOnly(newTableName));
+
+		if (tableExists(tablePath)) {
+			ObjectPath newPath = new ObjectPath(tablePath.getDatabaseName(), newTableName);
+
+			if (tableExists(newPath)) {
+				throw new TableAlreadyExistException(catalogName, newPath);
+			} else {
+				tables.put(newPath, tables.remove(tablePath));
+			}
+		} else if (!ignoreIfNotExists) {
+			throw new TableNotExistException(catalogName, tablePath);
+		}
+	}
+
+	@Override
+	public List<String> listAllTables(String databaseName) throws DatabaseNotExistException {
+		checkArgument(!StringUtils.isNullOrWhitespaceOnly(databaseName), "dbName cannot be null or empty");
+
+		if (!databaseExists(databaseName)) {
+			throw new DatabaseNotExistException(catalogName, databaseName);
+		}
+
+		return tables.keySet().stream()
+			.filter(k -> k.getDatabaseName().equals(databaseName)).map(k -> k.getObjectName())
+			.collect(Collectors.toList());
+	}
+
+	@Override
+	public List<String> listTables(String databaseName) throws DatabaseNotExistException {
```

Comment:
`listTables` -> `listTablesWithoutViews` or other better name.

Reply:
I used listAllTables to mean both, while listTables and listViews mean for each separately. This seems clear, especially with the javadoc. Would you rather use the long names?

Reply:
Hmm, by design I am sure about `listTables` should contain both tables and views. If you do not like the name for listTableWithoutViews, we can just remove this method, and add it later.(when we got the nice name).